### DISCUSSION TITLE
Defining hash: true in config breaks recipe

### DISCUSSION THREAD

**Discussion by Gytjarek** [chakra-ui/panda#2512] [Q&A] [answered]
I defined a button recipe: 
```
const buttonRecipe = defineRecipe({
  className: 'button',
  description: 'The styles for the Button component',
  base: {
    display: 'inline-flex',
  },
});
```

What works well, it also styles each button in my app with `button` class name. However, whenever I set `hash:true` in my config, it breaks this functionality; my HTML buttons no longer have styling. What am I doing wrong?

**Comment by anubra266** [comment]
@Gytjarek Could you create a reproduction with play.panda-css.com

**Reply by segunadebayo** [reply]
There's a pretty good chance you want to run `panda codegen` when you set `hash: true`

**Comment by Gytjarek** [comment]
> const buttonRecipe = defineRecipe({
>   className: 'button',
>   description: 'The styles for the Button component',
>   base: {
>     display: 'inline-flex',
>   },
> });

@anubra266 yes, created one: https://play.panda-css.com/Aj_Ylka4zi, so, if you go to the panda config and set `hash` to `true`, the button is no longer styled

**Reply by segunadebayo** [reply, marked as answer]
This is not a bug. You should not be hardcoding the className; instead, use the `button` function. When hash: true is set, the class name of the button is no longer "button" but rather a generated hash of that className.

https://play.panda-css.com/Aj_Ylka4zi/oTFJida6lo

![CleanShot 2024-04-18 at 17 10 53@2x](https://github.com/chakra-ui/panda/assets/6916170/66e8f2e9-0d53-4cf8-925a-814733561c6b)

**Reply by Gytjarek** [reply]
@segunadebayo  yea, that's clear, but what if I want to style buttons to which I don't have access? Like, I want to style each button in my app without specifying a button function. Is that possible?

**Reply by segunadebayo** [reply]
You could use the `staticCss` option and set the `hash: false`

It's currently not possible with the `hash` option.

**Reply by anubra266** [reply]
Also @Gytjarek I think you could just pass the `button()` call anywhere you'd have passed the `button` string.

**Reply by Gytjarek** [reply]
@anubra266 yeah, I don't have access to some buttons because they're being generated by third-party libraries

**Reply by anubra266** [reply]
@Gytjarek But you can pass a classname to those buttons right?
Maybe you could give a sample, lemme see If It's something I can figure out

**Reply by Gytjarek** [reply]
@anubra266 nope, I can not. This button has a 'button' class name by default, which is why I wanted to make it work for all buttons with this specific class name. I can still style it by defining CSS in a global style or create a CSS wrapper, but this means I would have to duplicate the code for the same element
